js/ui/file_manager/ui.file_manager.common.js

```diff
@@ -22,34 +30,31 @@ const whenSome = function(arg, onSuccess, onError) {
     };
 
     if(!Array.isArray(arg)) {
-        return when(arg)
-            .then(onSuccess,
-                error => {
-                    if(error) {
-                        onError(createErrorInfo(0, error));
-                    }
-                });
+        arg = [ arg ];
     }
 
     const deferreds = arg.map((item, index) => {
         return when(item)
-            .then(result => createResult(result, true),
+            .then(
+                result => {
+                    const resultObj = createResult(result, true);
+                    if(resultObj.error && onError) {
+                        onError(createErrorInfo(index, resultObj.error));
+                    }
+
+                    if(resultObj.success && onSuccess) {
```

Comment:
`resultObj.success` always is true, so the condition can be changed to `if(onSuccsess)`.
At summary, `resultObj` is not need at all, neither the `createResult()` function. Can we remove any of it?

Reply:
Fixed